cpp/src/arrow/io/file.cc

```diff
@@ -412,7 +412,8 @@ class MemoryMappedFile::MemoryMap : public MutableBuffer {
 
   bool closed() const { return !file_->is_open(); }
 
-  Status Open(const std::string& path, FileMode::type mode) {
+  Status Open(const std::string& path, FileMode::type mode, const int64_t length = 0,
```

Comment:
I would make `-1` the default length, so that we can tell if the user asked for 0-length region.